### DISCUSSION TITLE
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

### DISCUSSION THREAD

**Discussion by vladmandic** [vladmandic/face-api#134] [General]
[WebPack](https://webpack.js.org/) version 5 changed how it handles dynamic imports  

`FaceAPI` library itself does not use dynamic imports, but bundled **TensorFlow/JS** in `face-api.esm.js` includes some additional platform-specific capabilities  
For example, WASM binaries can be loaded using filesystem read file calls, but only when platform is *NodeJS*  

Presence of those dynamic capabilities detection calls causes WebPack to misinterpret type of a library used and generate warning messages:

```log
WARNING in ./node_modules/@vladmandic/human/dist/face-api.esm.js 13:11-18
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted
```

Note that this is NOT AN ERROR, but simply a `WebPack` warning that can be safely ignored.

However, if you want to eliminate such warnings,  
solution is to explicitly specify in `WebPack` configuration that `Human` is an ESM library:

in `webpack.config.js` add:

```js
module.exports = {
  module: {
    rules: [
      { test: /face-api.esm.js/, type: 'javascript/esm' },
    ],
  },
}
```

Note that WebPack is internally used by many frameworks and location of the config file may differ between frameworks (**React**, **Angular**, **Vue**, **Next**, etc.)  

To see how to modify WebPack configuration in cases when non-default configuration files are used, check framework specific documentation.

**Comment by ljnferreira** [comment]
Thanks for the presented solution, it worked fine for the problem I was trying to solve on my project!
